src/transformers/modeling_utils.py

```diff
@@ -1765,7 +1765,10 @@ def enable_input_require_grads(self):
         def make_inputs_require_grads(module, input, output):
             output.requires_grad_(True)
 
-        self._require_grads_hook = self.get_input_embeddings().register_forward_hook(make_inputs_require_grads)
+        if self.get_input_embeddings() is not None:
```

Comment:
`get_input_embeddings` shouldn't be `None` c.f. the docstring. What tests does this address?

Reply:
https://github.com/huggingface/transformers/blob/main/src/transformers/models/speecht5/modeling_speecht5.py#L2111
speecht5 case failure since it may return None

Reply:
FAILED tests/models/speecht5/test_modeling_speecht5.py::SpeechT5ModelTest::test_peft_gradient_checkpointing_enable_disable - AttributeError: 'NoneType' object has no attribute 'register_forward_hook'
FAILED tests/models/speecht5/test_modeling_speecht5.py::SpeechT5ForSpeechToSpeechTest::test_peft_gradient_checkpointing_enable_disable - AttributeError: 'NoneType' object has no attribute 'register_forward_hook'

Reply:
OK, this seems to be a speech t5 specific issue and sohuld be resolved but I wouldn't want this to block this PR. Could you: 

* Remove the `is not None` condition here
* Open an issue, linking to this PR and include the info you included here 
* Skip the test for SpeechT5 + add a link to the github issue in the skip message

Once done we can merge :) 

cc @ylacombe

Reply:
done